source/include/Events.h

```diff
@@ -0,0 +1,121 @@
+/**
+ 
+ Copyright (c) 2018 National Instruments Corp.
+ 
+ This software is subject to the terms described in the LICENSE.TXT file
+ 
+ SDG
+ */
+
+/*! \file
+ \brief Native Vireo event registration and handling
+  */
+
+#ifndef Events_h
+#define Events_h
+
+#include "TypeAndDataManager.h"
+
+namespace Vireo
+{
+
+enum {  // Event source
+    kEventSourceLVUserInt,
+    kEventSourceActiveXUserInt,
+    kEventSourceDotNetUserInt,
+    kEventSourceUserEvent = 25 };
+
+enum {
+    kEventTypeNull = 0,
+    kEventTypeTimeout = 1,
+    kEventTypeValueChange = 2,
+    kEventTypePanelClose_ =  3,
+    kEventTypeAppClose_ = 4,
+    kEventTypeMenuSelectedUser = 5,
+    kEventTypeMenuSelectedApp_ = 6,
+    kEventTypeMouseDown_ = 7,
+    kEventTypeMouseUp = 8,
+    kEventTypeMouseMove = 9,
+    kEventTypeKeyDown_ = 10,
+    kEventTypeKeyRepeat_ = 11,
+    kEventTypeKeyUp = 12,
+    kEventTypeListBoxDblClk = 13,
+    kEventTypeMouseLeave = 14,
+    kEventTypeMouseEnter = 15,
+    kEventTypeTreeItemOpen_ = 16,
+    kEventTypeTreeItemClose_ = 17,
+    kEventTypeMouseDown = 18,
+    kEventTypePanelResize = 19,
+    kEventTypeTreeDrop_ = 20,
+    kEventTypeTreeDrag_ = 21,
+    kEventTypeTreeDblClk = 22,
+    kEventTypeKeyDown = 23,
+    kEventTypeKeyRepeat =  24,
+    kEventTypePanelClose = 25,
+    kEventTypeAppClose = 26,
+    kEventTypeMenuSelectedApp =  27,
+    // ...
+    kEventTypeUserEvent = 1000
+};
+
+typedef UInt32 EventSource;
+typedef UInt32 EventType;
+
+typedef UInt32 EventControlUID;  // TBD
+enum { kNotAnEventControlUID = 0 };
+
+struct EventCommonData {
+    // Common fields
+    EventSource eventSource;  // Event source and type are used internally but not visible in Event structure in NXG
```

Comment:
Why don't we create a separate structure for what goes in VIA based on what is needed in NXG. If it is not visible in the VI, DFIR has to still emit something for this. I think that will be confusing.

Reply:
Confusing to who, people writing Via code manually?   It's much easier to have the cluster explicitly allocated and match the event data which has to be there, even though it isn't visible on the user's G diagram.  It's there in Desktop/NCG too.